src/main/java/org/apache/sysml/runtime/instructions/spark/WriteSPInstruction.java

```diff
@@ -103,6 +104,13 @@ public static WriteSPInstruction parseInstruction ( String str )
 			
 			boolean isInputMB = Boolean.parseBoolean(parts[7]);
 			inst.setInputMatrixBlock(isInputMB);
+		} else {
+			FileFormatProperties ffp = new FileFormatProperties();
+			String description = parts[4];
```

Comment:
Since the description can be a variable, at parse time you should do parse part[4] as CPOperand. Then in process instruction, you should check if the passed value is a variable or a literal. For example:
```bash
String fname = ec.getScalarInput(input2.getName(), ValueType.STRING, input2.isLiteral()).getStringValue();
```